src/boost_histogram/_internal/axis.py

```diff
@@ -525,12 +531,14 @@ def __init__(self, categories, **kwargs):
         # Python interfaces protect against that
 
         if options == {"growth"}:
-            self._ax = ca.category_str_growth(tuple(categories), metadata)
+            self._ax = ca.category_str_growth(tuple(categories))
         elif options == set():
-            self._ax = ca.category_str(tuple(categories), metadata)
+            self._ax = ca.category_str(tuple(categories))
         else:
             raise KeyError("Unsupported collection of options")
 
+        self.metadata = metadata
```

Comment:
It looks like I can still set metadata to any type here. I am personally fine with removing the metadata keyword from init, it is a breaking change, however.

Reply:
I originally allowed both, but I think (as mentioned above) it now is truly always a dict, so I can simplify a bit! Metadata is allowed as a kwarg and is printed in the repr for backward compatibility, but we could remove that if you want. Since Boost.Histogram has a metadata slot, I think it's okay to allow easy access to ax.metadata this way?

Reply:
How about this: all unrecognized kwargs are added to the dict? This means you can write `bh.axis.Regular(..., name="foo", title="bar")` and the old `bh.axis.Regular(..., metadata="foo")` would still work, but metadata is now just one of those unrecognized kwargs.

Reply:
If a keyword conflicts with an axis method, the keyword needs to be rejected, however.

Reply:
I don't think I like that idea, for three reasons: a) mixing keyword arguments and `**kwargs` is ugly; if you mistype a keyword argument (`grow` instead of `growth`, for example), instead of immediately complaining loudly, it silently works but does the wrong thing (saves `.grow=True` and doesn't activate growth); b) you can't add any keyword arguments later without changing the API (or in downstream libraries like Hist; if we add a `flow=True` shortcut that would cause two different behaviors), and c) changing axis classes will not cause unused keywords to throw an error, but will store them instead. For example, setting `circular=True` on a `IntCategory` would silently "work" (different form of point a).  You can also always turn this on later, but you can't remove it once added. The fact you could set title/name is a plus. Having a tight API on the upstream library is likely better?

Reply:
I'll go ahead and merge this, since it's blocking further progress. Please let me know if you want **kwargs (or any other changes), and I'll get them in (in a separate PR).